drupal/sites/default/boinc/modules/boincuser/boincuser.module

```diff
@@ -1546,9 +1546,18 @@ function boincuser_create_account() {
       if ($boinc_user) {
         // Cross reference Drupal account with BOINC.
         $reference = db_query("INSERT INTO {boincuser} SET uid=%d, boinc_id=%d", $user->uid, $boinc_user->id);
+
         if (!$reference) {
           xml_error(-1, 'error connecting BOINC account to Drupal');
         }
+
+        // if terms of use exist, the user must have opted-in.
+        // @todo - test this block of code!
+        $termsofuse = variable_get('boinc_weboptions_termsofuse', '');
```

Comment:
~~~This seems to be a Drupal-specific RPC customization. If I'm right, what's the point of it as long as BOINC's stock RPC isn't changed the same way. We lose API/RPC coherence for the service consumers like the BOINC client (or account managers) that would have to be updated as well to make use of that new parameter.~~~

See below, I think misinterpreted the variable and the comment above it. Please confirm: this means when this RPC gets called **and** the terms of use exist, there's no way the user can't have consented - so let's store that fact in the DB.